tools/pipelines/templates/include-publish-docker-service-steps.yml

```diff
@@ -71,9 +71,20 @@ jobs:
               ${{ parameters.containerRegistryUrl }}/${{ parameters.containerName }}:latest,
               ${{ parameters.containerRegistryUrl }}/${{ parameters.containerName }}:$(containerTagSuffix)
 
-        - template: /tools/pipelines/templates/include-git-tag-steps.yml@self
-          parameters:
-            tagName: ${{ parameters.tagName }}
+- job: TagRelease
+  displayName: Tag Release
+  # Note: must be kept in sync with the name of the deployment job above
+  dependsOn: publish_${{ replace(parameters.environment, '-', '_') }}
+  # Only tag the repo if a tag name is provided.
+  condition: and(succeeded(), ne('${{ parameters.tagName }}', ''))
+  steps:
+  - checkout: self
+    clean: true
+  - template: /tools/pipelines/templates/include-git-tag-steps.yml@self
+    parameters:
+      tagName: ${{ parameters.tagName }}
+      version: $[ stageDependencies.build.build.outputs['SetVersion.version']]
```

Comment:
I don't think this is going to work. Only `${{ }}` syntax (compile-time expression) is replaced when parsing the file (plus, since this value is the output of a job, it only exists at runtime), so this string would be passed verbatim, end up in the script as `tag=<interpolated tag name>}_v$[ stageDependencies.build.build.outputs['SetVersion.version']]`, and the `$[ ]` syntax (runtime expression) won't be interpolated in that case, they can only be used in `variables:` blocks and `condition:` lines.

What we can do is add this bit in the `variables:` of the job(s) that end up executing the `include-git-tag-steps.yml` template, and then we can keep `$(version)` in the script. That's how the deployment job that publishes the packages to npm is doing it.

Reply:
The thing I don't like about your proposed solution is that it is no longer self-contained. Now, the include-git-tag-steps cannot be used anywhere - it can only be used where stages with specific names and outputs exist in the including pipeline's "ancestry".

Reply:
It's also weird to me that one variable is passed one way, and the other is passed another way.

Reply:
Fair. We can keep the parameter, pass `$(version)` as its "value" (at compile-time this won't get interpolated, so the final yml will read `tag=<interpolated tag name>_v$(version)` ), and we still want the new entry in the job's variables so `$(version)` exists at runtime. It feels a bit unnatural, but I think that might be the only way to leverage the output of a previous job when the one that consumes it needs to pass it when including templates.

Reply:
>It's also weird to me that one variable is passed one way, and the other is passed another way.

It totally is. One of the biggest pitfalls of ADO in my opinion, and pretty easy to make mistakes with it. Parameters and variables are separate things with very different behaviors and lifetimes, but they're so easy to mix incorrectly.

Reply:
Updated to set variables at the job level for version, but tagName remains a parameter.